clients/drcachesim/reader/config_reader.cpp

```diff
@@ -60,54 +60,83 @@ config_reader_t::configure(const string &config_file,
     // Walk through the configuration file.
     while (!fin.eof()) {
         string param;
-        fin >> ws >> param;
+
+        if (!(fin >> ws >> param)) {
+            ERRMSG("Unable to read from the configuration file\n");
+            return false;
+        }
 
         if (param == "//") {
             // A comment.
-            getline(fin, param);
+            if (!getline(fin, param)) {
+                ERRMSG("Comment expected but not found\n");
+                return false;
+            }
         }
         else if (param == "num_cores") {
             // Number of cache cores.
-            fin >> knobs.num_cores;
+            if (!(fin >> knobs.num_cores)) {
+                ERRMSG("Unable to read from the configuration file\n");
```

Comment:
suggest: It will be much nicer for the user if the error message includes where in the file the problem is: so "Error reading num_cores value from the configuration file" or something.  Here and below.

Reply:
Done.